src/GovUk.Education.ExploreEducationStatistics.Publisher/Services/ReleaseService.cs

```diff
@@ -125,15 +145,29 @@ public async Task SetPublishedDatesAsync(Guid id, DateTime published)
                 await _statisticsDbContext.SaveChangesAsync();
             }
         }
-
-        private static bool IsReleasePublished(Release release, IEnumerable<Guid> includedReleaseIds)
+        
+        public List<ReleaseFileReference> GetReleaseFileReferences(Guid releaseId, params ReleaseFileTypes[] types)
         {
-            return release.Live || includedReleaseIds.Contains(release.Id);
+            return _contentDbContext
+                .ReleaseFiles
+                .Include(rf => rf.ReleaseFileReference)
+                .Where(rfr => rfr.ReleaseId == releaseId)
+                .Select(rf => rf.ReleaseFileReference)
+                .Where(rfr => types.Contains(rfr.ReleaseFileType))
+                .ToList();
         }
 
-        private static bool IsLatestVersionOfRelease(Publication publication, Guid releaseId)
+        public async Task RemoveDataForPreviousVersions(IEnumerable<Guid> releaseIds)
```

Comment:
So we're unlinking the Subjects from the original Release version in the stats db?  For what purpose is this?

Reply:
So the public-stats db unlike the statistics admin db should only ever hold data for the amendment & not previous versions of it - I can't test this properly until it's on dev as it stands.